pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -170,6 +170,9 @@ func AddToManager(mgr manager.Manager, r reconcile.Reconciler) error {
 		IsController: true,
 		OwnerType:    &hivev1.ClusterDeployment{},
 	})
+	if err != nil {
+		return err
```

Comment:
I suspect all of these should have a log.WithError().Error(), otherwise we're probably not getting any info in the logs.

Reply:
In this case, the controller wouldn't start so we'd know something's up but yeah, a log statement wouldn't hurt

Reply:
@csrwng @dgoodwin I search through the whole Hive code base and no where we have used `log.WithError().Error()` in the `func AddToManager(mgr manager.Manager, r reconcile.Reconciler) error`.  I also could not verify about what error `Watch` returns. So if you guys think we should add logs then we need to add it for 
```
pkg/controller/clusterdeprovisionrequest/clusterdeprovisionrequest_controller.go:
pkg/controller/clusterversion/clusterversion_controller.go:
pkg/controller/controlplanecerts/controlplanecerts_controller.go:
pkg/controller/dnszone/dnszone_controller.go:
pkg/controller/installlogmonitor/installlogmonitor_controller.go:
pkg/controller/remoteingress/remoteingress_controller.go:
pkg/controller/remotemachineset/remotemachineset_controller.go:
pkg/controller/syncidentityprovider/syncidentityprovider_controller.go:
pkg/controller/syncset/syncset_controller.go:
pkg/controller/syncsetinstance/syncsetinstance_controller.go:
pkg/controller/unreachable/unreachable_controller.go:
pkg/operator/hive/hive_controller.go:
```

Reply:
@dgoodwin @csrwng Let me know your thoughts on this.